src/function/utils/isPrime.js

```diff
@@ -59,12 +59,52 @@ export const createIsPrime = /* #__PURE__ */ factory(name, dependencies, ({ type
       if (n.lte(3)) return n.gt(1)
       if (n.mod(2).eq(0) || n.mod(3).eq(0)) return false
 
-      for (let i = 5; n.gte(i * i); i += 6) {
-        if (n.mod(i).eq(0) || n.mod(i + 2).eq(0)) {
-          return false
+      function modPow (base, exponent, modulus) {
+        // exponent can be huge, use non-recursive variant
+        let accumulator = 1
+        while (!exponent.eq(0)) {
+          if (exponent.mod(2).eq(0)) {
+            exponent = exponent.div(2)
+            base = base.mul(base).mod(modulus)
+          } else {
+            exponent = exponent.sub(1)
+            accumulator = base.mul(accumulator).mod(modulus)
+          }
         }
+        return accumulator
       }
 
+      // https://en.wikipedia.org/wiki/Miller%E2%80%93Rabin_primality_test#Deterministic_variants
+      const Decimal = n.constructor.clone({ precision: n.constructor.precision * 2 })
+      n = new Decimal(n)
+      let r = 0
+      let d = n.sub(1)
+      while (d.mod(2).eq(0)) {
+        d = d.div(2)
+        r += 1
+      }
+      let bases = null
+      // https://en.wikipedia.org/wiki/Miller–Rabin_primality_test#Testing_against_small_sets_of_bases
+      if (n.lt('3317044064679887385961981')) {
+        bases = [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41].filter(x => x < n)
+      } else {
+        const max = Math.min(n.toNumber() - 2, Math.floor(2 * Math.pow(n.decimalPlaces() * Math.log(10), 2)))
```

Comment:
For this calculation of the max (or min?), you transform `Decimal`  into a regular number. Should these calculations be done as `Decimal`?

Reply:
it should work with conversion to a number even when it will be converted to Infinity as the second part should be small

Reply:
Ah yes you're right, then it works just fine indeed 👍